### ISSUE TITLE
Clippy lints only applying in `main.rs`

### ISSUE THREAD

**Issue by cameron1024** [rust-lang/rust-clippy#7549]
I'm not sure if this is actually a bug or if I'm doing something wrong, if I create a blank project with `cargo new example`

```rust
// main.rs
#![deny(clippy::print_stdout)]
mod other;
fn main() {
  println!("Hello, world!");
  crate::other::also_prints();
}

// other.rs
pub fn also_prints() {
    println!("nothing wrong here")
}
```
I expect the rule to apply to both files and deny both `println!()` calls.

Instead, only the `println!()` in `main.rs` is highlighted as an error. If I comment out that line and run with `cargo run`, the program compiles and runs fine (behaviour is consistent between VSCode error highlighting and `cargo clippy`).

This doesn't seem to happen with regular lints, e.g. if I add `'#![deny(unsafe_code)]`, an `unsafe` block in either file shows an error.

Is there some flag I need to pass in somehow to tell clippy I mean "apply this rule to my entire crate"?

### Meta

- `cargo clippy -V`: e.g. clippy 0.1.54 (a178d03 2021-07-26)
- `rustc -Vv`:
```
rustc 1.54.0 (a178d0322 2021-07-26)
binary: rustc
commit-hash: a178d0322ce20e33eac124758e837cbd80a6f633
commit-date: 2021-07-26
host: x86_64-unknown-linux-gnu
release: 1.54.0
LLVM version: 12.0.1
```

**Comment by giraffate** [comment]
You can globally enable or disable lints by passing flags like:
```
cargo clippy -- -D clippy::print_stdout
```

`#!` is the inner-attribute, it's applied in item declared within, so it won't work in `other` module in your example.

**Comment by mikerite** [comment]
This is a duplicate of #5721. The problem is that the `print_stdout` is running in a pre-expansion pass. Pre-expansion passes are broken and won't be fixed. This lints and others need to be rewritten to use normal passes.

**Comment by giraffate** [comment]
Oh, thanks! https://github.com/rust-lang/rust-clippy/issues/7549#issuecomment-895704518 is right.

**Comment by cameron1024** [comment]
Thanks 😀

**Comment by jdonszelmann** [comment]
fixed, this is no longer a pre-expansion lint (it's late now) (@y21)
